crates/interpreter/src/interpreter/memory.rs

```diff
@@ -110,6 +109,15 @@ impl Memory {
             *i = 0;
         }
     }
+    /// In memory copy given a src, dst, and offset
+    /// Assumes that we check that src + offset > self.data.len()
+    #[inline(always)]
+    pub fn copy(&mut self, src: usize, dst: usize, offset: usize) {
+        if src + offset > self.data.len() {
+            return;
+        }
+        self.data.copy_within(src..src + offset, dst);
+    }
```

Comment:
Few nits. wouth call it `lenght` and would remove this `if` as we already resized memory before hand.

```suggestion
    /// In memory copy given a src, dst, and length
    ///
    /// # Safety
    /// The caller is responsible to check that we resized memory properly.
    #[inline(always)]
    pub fn copy(&mut self, dst: usize, src: usize, length: usize) {
        self.data.copy_within(src..src + length, dst);
    }
```